src/utils/ConvertToEmbeddableUrl.tsx

```diff
@@ -8,28 +15,55 @@ export interface ConvertToEmbeddableUrlOptions {
 // @debt Replace the naive logic with use of new URL(url) which provides standard parsing before doing checks
 export const convertToEmbeddableUrl: (
   options: ConvertToEmbeddableUrlOptions
-) => string = ({ url, autoPlay = DEFAULT_VENUE_AUTOPLAY }) => {
-  if (url?.includes("youtube")) {
-    url = url?.replace("watch?v=", "embed/");
-  } else if (
-    url?.includes("vimeo") &&
-    !url?.includes("player") &&
-    // NOTE: If you have a scheduled live event, it gives you a different embed code
-    !url?.includes("vimeo.com/event")
+) => string = ({ url, autoPlay }) => {
+  if (!url) {
+    return "";
+  }
+
+  let urlObj = new URL(url);
+  const { host, searchParams, pathname } = urlObj;
+  const youtubeVideoId = searchParams.get("v") || pathname.slice(1);
+  const twitchVideoId = searchParams.get("video");
+  const twitchId = searchParams.get("channel") || twitchVideoId;
+  const isTwitch = host.includes(TWITCH_SHORT_URL);
+
+  if (
+    host?.includes(YOUTUBE_SHORT_URL_STRING) &&
+    !pathname?.includes("embed")
   ) {
-    url = url?.replace("vimeo.com/", "player.vimeo.com/video/");
+    urlObj = new URL(`${YOUTUBE_EMBED_URL}${youtubeVideoId}`);
+    const params = searchParams.toString().split("&");
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams no need for `.toString()` there is `.getAll()` and similar accessors on the search parameters object

Reply:
`URLSearchParams.getAll(name)` requires a `name` prop to be passed and returns all mentions of that same prop
In this piece of code I'm extracting all props from the parameters
I've been looking for an api method that would do the same but without this workaround, but failed to find one :(